src/components/__tests__/LeaderboardTable.spec.js

```diff
@@ -0,0 +1,35 @@
+import { describe, it, expect } from "vitest";
+import { mount } from "@vue/test-utils";
+import { createVuetify } from "vuetify";
+import LeaderboardTable from "../LeaderboardTable.vue";
+
+describe("LeaderboardTable", () => {
+  const vuetify = createVuetify();
+  const profileBaseUrl = "https://meta.wikimedia.org/wiki/User:";
+  it("renders properly", () => {
+    const LeaderboardTableProps = {
+      contributions: [
+        {
+          userName: "Hannah Waruguru",
+          toolsEdited: 12,
+        },
+      ],
+      title: "All Time Best",
+      profileBaseUrl,
+    };
+    const wrapper = mount(LeaderboardTable, {
+      props: LeaderboardTableProps,
+      global: {
+        plugins: [vuetify],
+      },
+    });
+    expect(wrapper.text()).toContain("All Time Best");
+
+    const firstRow = wrapper.findAll('td')[0].find('a')
+    expect(firstRow.text()).toBe('Hannah Waruguru');
+    expect(firstRow.attributes().href).toBe('https://meta.wikimedia.org/wiki/User:Hannah_Waruguru')
+
```

Comment:
I think we should also assert that the number of rows rendered in the table is equal to the length of the contributions array that was passed in.

Reply:
Great @Damilare1 let me work on this

Reply:
Hi Hannah, thanks for the changes. We are almost there but I noticed a few things. For example instead of running `findAll("td")` on line 28 and 34, we could simply cache that result in a variable as shown in the snippet below. 

Some other points to note:

- Be specific with the search in order to not get unrelated items in the results.
- I'd also prefer using the array length value in the length assertion rather than an hardcoded value.

Here's what I'm talking about:

```
    const tbody = wrapper.find("tbody");
    const rows = tbody.findAll("tr");

    // Assert that the table rows is equal to the number of items in the contributions array
    expect(rows.length).toBe(LeaderboardTableProps.contributions.length);
    
    const firstRowCols = rows[0].findAll("td");

    // Assert that the anchor text containing the username is present as a child of the first cell in the row
    // and it renders the correct text.
    const firstCol = firstRowCols[0].find("a");
    expect(firstCol.text()).toBe("Hannah Waruguru");
    expect(firstCol.attributes().href).toBe(
      "https://meta.wikimedia.org/wiki/User:Hannah_Waruguru"
    );

    // Assert that the text containing the contribution number for the user is in the second cell in the row
    const contributionCol = firstRowCols[1];
    expect(contributionCol.text()).toBe("12");
```